### ISSUE TITLE
Undefined column 'AnyKey'

### ISSUE THREAD

**Issue by netniV** [Cacti/plugin_routerconfigs#90]
Hi, 

After the check it seems that it's not able to find the prompt, with this error : 

```
2019/02/19 15:47:18 - ERROR PHP NOTICE in Plugin 'routerconfigs': Undefined index: AnyKey in file: /usr/share/cacti/plugins/routerconfigs/classes/PHPConnection.php on line: 284
2019/02/19 15:47:33 - RCONFIG 10.234.254.14 (SSH) -> DEBUG: -->
2019/02/19 15:47:33 - RCONFIG 10.234.254.14 (SSH) -> DEBUG: Attempt 3 of 10 to find prompt
2019/02/19 15:47:33 - RCONFIG 10.234.254.14 (SSH) -> DEBUG: Timeout of 5 seconds has been reached
2019/02/19 15:47:28 - CMDPHP PHP ERROR NOTICE Backtrace: (/plugins/routerconfigs/router-download.php: 133 plugin_routerconfigs_download)(/plugins/routerconfigs/functions.php: 159 plugin_routerconfigs_download_config)(/plugins/routerconfigs/functions.php: 499 EnsureEnabled)(/plugins/routerconfigs/classes/PHPConnection.php: 134 DoCommand)(/plugins/routerconfigs/classes/PHPConnection.php: 229 GetResponse)(/plugins/routerconfigs/classes/PHPConnection.php: 284 CactiErrorHandler)(/lib/functions.php: 4590 cacti_debug_backtrace)
2019/02/19 15:47:28 - ERROR PHP NOTICE in Plugin 'routerconfigs': Undefined index: AnyKey in file: /usr/share/cacti/plugins/routerconfigs/classes/PHPConnection.php on line: 284
2019/02/19 15:47:28 - RCONFIG 10.234.254.14 (SSH) -> DEBUG: -->
2019/02/19 15:47:28 - RCONFIG 10.234.254.14 (SSH) -> DEBUG: Attempt 2 of 10 to find prompt
2019/02/19 15:47:28 - RCONFIG 10.234.254.14 (SSH) -> DEBUG: Timeout of 5 seconds has been reached
2019/02/19 15:47:23 - CMDPHP PHP ERROR NOTICE Backtrace: (/plugins/routerconfigs/router-download.php: 133 plugin_routerconfigs_download)(/plugins/routerconfigs/functions.php: 159 plugin_routerconfigs_download_config)(/plugins/routerconfigs/functions.php: 499 EnsureEnabled)(/plugins/routerconfigs/classes/PHPConnection.php: 134 DoCommand)(/plugins/routerconfigs/classes/PHPConnection.php: 229 GetResponse)(/plugins/routerconfigs/classes/PHPConnection.php: 284 CactiErrorHandler)(/lib/functions.php: 4590 cacti_debug_backtrace)
2019/02/19 15:47:23 - ERROR PHP NOTICE in Plugin 'routerconfigs': Undefined index: AnyKey in file: /usr/share/cacti/plugins/routerconfigs/classes/PHPConnection.php on line: 284
2019/02/19 15:47:23 - RCONFIG 10.234.254.14 (SSH) -> DEBUG: -->
2019/02/19 15:47:23 - RCONFIG 10.234.254.14 (SSH) -> DEBUG: Attempt 1 of 10 to find prompt
2019/02/19 15:47:23 - RCONFIG 10.234.254.14 (SSH) -> DEBUG: Timeout of 5 seconds has been reached
2019/02/19 15:47:18 - CMDPHP PHP ERROR NOTICE Backtrace: (/plugins/routerconfigs/router-download.php: 133 plugin_routerconfigs_download)(/plugins/routerconfigs/functions.php: 159 plugin_routerconfigs_download_config)(/plugins/routerconfigs/functions.php: 499 EnsureEnabled)(/plugins/routerconfigs/classes/PHPConnection.php: 134 DoCommand)(/plugins/routerconfigs/classes/PHPConnection.php: 229 GetResponse)(/plugins/routerconfigs/classes/PHPConnection.php: 284 CactiErrorHandler)(/lib/functions.php: 4590 cacti_debug_backtrace)
2019/02/19 15:47:18 - ERROR PHP NOTICE in Plugin 'routerconfigs': Undefined index: AnyKey in file: /usr/share/cacti/plugins/routerconfigs/classes/PHPConnection.php on line: 284
2019/02/19 15:47:18 - CMDPHP PHP ERROR NOTICE Backtrace: (/plugins/routerconfigs/router-download.php: 133 plugin_routerconfigs_download)(/plugins/routerconfigs/functions.php: 159 plugin_routerconfigs_download_config)(/plugins/routerconfigs/functions.php: 499 EnsureEnabled)(/plugins/routerconfigs/classes/PHPConnection.php: 134 DoCommand)(/plugins/routerconfigs/classes/PHPConnection.php: 229 GetResponse)(/plugins/routerconfigs/classes/PHPConnection.php: 284 CactiErrorHandler)(/lib/functions.php: 4590 cacti_debug_backtrace)
2019/02/19 15:47:18 - ERROR PHP NOTICE in Plugin 'routerconfigs': Undefined index: AnyKey in file: /usr/share/cacti/plugins/routerconfigs/classes/PHPConnection.php on line: 284
2019/02/19 15:47:18 - CMDPHP PHP ERROR NOTICE Backtrace: (/plugins/routerconfigs/router-download.php: 133 plugin_routerconfigs_download)(/plugins/routerconfigs/functions.php: 159 plugin_routerconfigs_download_config)(/plugins/routerconfigs/functions.php: 499 EnsureEnabled)(/plugins/routerconfigs/classes/PHPConnection.php: 134 DoCommand)(/plugins/routerconfigs/classes/PHPConnection.php: 229 GetResponse)(/plugins/routerconfigs/classes/PHPConnection.php: 284 CactiErrorHandler)(/lib/functions.php: 4590 cacti_debug_backtrace)
2019/02/19 15:47:18 - ERROR PHP NOTICE in Plugin 'routerconfigs': Undefined index: AnyKey in file: /usr/share/cacti/plugins/routerconfigs/classes/PHPConnection.php on line: 284
2019/02/19 15:47:18 - CMDPHP PHP ERROR NOTICE Backtrace: (/plugins/routerconfigs/router-download.php: 133 plugin_routerconfigs_download)(/plugins/routerconfigs/functions.php: 159 plugin_routerconfigs_download_config)(/plugins/routerconfigs/functions.php: 499 EnsureEnabled)(/plugins/routerconfigs/classes/PHPConnection.php: 134 DoCommand)(/plugins/routerconfigs/classes/PHPConnection.php: 229 GetResponse)(/plugins/routerconfigs/classes/PHPConnection.php: 284 CactiErrorHandler)(/lib/functions.php: 4590 cacti_debug_backtrace)
2019/02/19 15:47:18 - ERROR PHP NOTICE in Plugin 'routerconfigs': Undefined index: AnyKey in file: /usr/share/cacti/plugins/routerconfigs/classes/PHPConnection.php on line: 284
2019/02/19 15:47:18 - CMDPHP PHP ERROR NOTICE Backtrace: (/plugins/routerconfigs/router-download.php: 133 plugin_routerconfigs_download)(/plugins/routerconfigs/functions.php: 159 plugin_routerconfigs_download_config)(/plugins/routerconfigs/functions.php: 499 EnsureEnabled)(/plugins/routerconfigs/classes/PHPConnection.php: 134 DoCommand)(/plugins/routerconfigs/classes/PHPConnection.php: 229 GetResponse)(/plugins/routerconfigs/classes/PHPConnection.php: 284 CactiErrorHandler)(/lib/functions.php: 4590 cacti_debug_backtrace)
2019/02/19 15:47:18 - ERROR PHP NOTICE in Plugin 'routerconfigs': Undefined index: AnyKey in file: /usr/share/cacti/plugins/routerconfigs/classes/PHPConnection.php on line: 284
2019/02/19 15:47:18 - CMDPHP PHP ERROR NOTICE Backtrace: (/plugins/routerconfigs/router-download.php: 133 plugin_routerconfigs_download)(/plugins/routerconfigs/functions.php: 159 plugin_routerconfigs_download_config)(/plugins/routerconfigs/functions.php: 499 EnsureEnabled)(/plugins/routerconfigs/classes/PHPConnection.php: 134 DoCommand)(/plugins/routerconfigs/classes/PHPConnection.php: 229 GetResponse)(/plugins/routerconfigs/classes/PHPConnection.php: 284 CactiErrorHandler)(/lib/functions.php: 4590 cacti_debug_backtrace)
2019/02/19 15:47:18 - ERROR PHP NOTICE in Plugin 'routerconfigs': Undefined index: AnyKey in file: /usr/share/cacti/plugins/routerconfigs/classes/PHPConnection.php on line: 284
2019/02/19 15:47:18 - CMDPHP PHP ERROR NOTICE Backtrace: (/plugins/routerconfigs/router-download.php: 133 plugin_routerconfigs_download)(/plugins/routerconfigs/functions.php: 159 plugin_routerconfigs_download_config)(/plugins/routerconfigs/functions.php: 499 EnsureEnabled)(/plugins/routerconfigs/classes/PHPConnection.php: 134 DoCommand)(/plugins/routerconfigs/classes/PHPConnection.php: 229 GetResponse)(/plugins/routerconfigs/classes/PHPConnection.php: 284 CactiErrorHandler)(/lib/functions.php: 4590 cacti_debug_backtrace)
2019/02/19 15:47:18 - ERROR PHP NOTICE in Plugin 'routerconfigs': Undefined index: AnyKey in file: /usr/share/cacti/plugins/routerconfigs/classes/PHPConnection.php on line: 284
2019/02/19 15:47:18 - CMDPHP PHP ERROR NOTICE Backtrace: (/plugins/routerconfigs/router-download.php: 133 plugin_routerconfigs_download)(/plugins/routerconfigs/functions.php: 159 plugin_routerconfigs_download_config)(/plugins/routerconfigs/functions.php: 499 EnsureEnabled)(/plugins/routerconfigs/classes/PHPConnection.php: 134 DoCommand)(/plugins/routerconfigs/classes/PHPConnection.php: 229 GetResponse)(/plugins/routerconfigs/classes/PHPConnection.php: 284 CactiErrorHandler)(/lib/functions.php: 4590 cacti_debug_backtrace)
2019/02/19 15:47:18 - ERROR PHP NOTICE in Plugin 'routerconfigs': Undefined index: AnyKey in file: /usr/share/cacti/plugins/routerconfigs/classes/PHPConnection.php on line: 284
2019/02/19 15:47:18 - CMDPHP PHP ERROR NOTICE Backtrace: (/plugins/routerconfigs/router-download.php: 133 plugin_routerconfigs_download)(/plugins/routerconfigs/functions.php: 159 plugin_routerconfigs_download_config)(/plugins/routerconfigs/functions.php: 499 EnsureEnabled)(/plugins/routerconfigs/classes/PHPConnection.php: 134 DoCommand)(/plugins/routerconfigs/classes/PHPConnection.php: 229 GetResponse)(/plugins/routerconfigs/classes/PHPConnection.php: 284 CactiErrorHandler)(/lib/functions.php: 4590 cacti_debug_backtrace)
2019/02/19 15:47:18 - ERROR PHP NOTICE in Plugin 'routerconfigs': Undefined index: AnyKey in file: /usr/share/cacti/plugins/routerconfigs/classes/PHPConnection.php on line: 284
2019/02/19 15:47:18 - CMDPHP PHP ERROR NOTICE Backtrace: (/plugins/routerconfigs/router-download.php: 133 plugin_routerconfigs_download)(/plugins/routerconfigs/functions.php: 159 plugin_routerconfigs_download_config)(/plugins/routerconfigs/functions.php: 499 EnsureEnabled)(/plugins/routerconfigs/classes/PHPConnection.php: 134 DoCommand)(/plugins/routerconfigs/classes/PHPConnection.php: 229 GetResponse)(/plugins/routerconfigs/classes/PHPConnection.php: 284 CactiErrorHandler)(/lib/functions.php: 4590 cacti_debug_backtrace)
2019/02/19 15:47:18 - ERROR PHP NOTICE in Plugin 'routerconfigs': Undefined index: AnyKey in file: /usr/share/cacti/plugins/routerconfigs/classes/PHPConnection.php on line: 284
2019/02/19 15:47:18 - CMDPHP PHP ERROR NOTICE Backtrace: (/plugins/routerconfigs/router-download.php: 133 plugin_routerconfigs_download)(/plugins/routerconfigs/functions.php: 159 plugin_routerconfigs_download_config)(/plugins/routerconfigs/functions.php: 499 EnsureEnabled)(/plugins/routerconfigs/classes/PHPConnection.php: 134 DoCommand)(/plugins/routerconfigs/classes/PHPConnection.php: 229 GetResponse)(/plugins/routerconfigs/classes/PHPConnection.php: 284 CactiErrorHandler)(/lib/functions.php: 4590 cacti_debug_backtrace)
2019/02/19 15:47:18 - ERROR PHP NOTICE in Plugin 'routerconfigs': Undefined index: AnyKey in file: /usr/share/cacti/plugins/routerconfigs/classes/PHPConnection.php on line: 284
2019/02/19 15:47:18 - CMDPHP PHP ERROR NOTICE Backtrace: (/plugins/routerconfigs/router-download.php: 133 plugin_routerconfigs_download)(/plugins/routerconfigs/functions.php: 159 plugin_routerconfigs_download_config)(/plugins/routerconfigs/functions.php: 499 EnsureEnabled)(/plugins/routerconfigs/classes/PHPConnection.php: 134 DoCommand)(/plugins/routerconfigs/classes/PHPConnection.php: 229 GetResponse)(/plugins/routerconfigs/classes/PHPConnection.php: 284 CactiErrorHandler)(/lib/functions.php: 4590 cacti_debug_backtrace)
2019/02/19 15:47:18 - ERROR PHP NOTICE in Plugin 'routerconfigs': Undefined index: AnyKey in file: /usr/share/cacti/plugins/routerconfigs/classes/PHPConnection.php on line: 284
2019/02/19 15:47:18 - CMDPHP PHP ERROR NOTICE Backtrace: (/plugins/routerconfigs/router-download.php: 133 plugin_routerconfigs_download)(/plugins/routerconfigs/functions.php: 159 plugin_routerconfigs_download_config)(/plugins/routerconfigs/functions.php: 499 EnsureEnabled)(/plugins/routerconfigs/classes/PHPConnection.php: 134 DoCommand)(/plugins/routerconfigs/classes/PHPConnection.php: 229 GetResponse)(/plugins/routerconfigs/classes/PHPConnection.php: 284 CactiErrorHandler)(/lib/functions.php: 4590 cacti_debug_backtrace)
2019/02/19 15:47:18 - ERROR PHP NOTICE in Plugin 'routerconfigs': Undefined index: AnyKey in file: /usr/share/cacti/plugins/routerconfigs/classes/PHPConnection.php on line: 284
2019/02/19 15:47:18 - CMDPHP PHP ERROR NOTICE Backtrace: (/plugins/routerconfigs/router-download.php: 133 plugin_routerconfigs_download)(/plugins/routerconfigs/functions.php: 159 plugin_routerconfigs_download_config)(/plugins/routerconfigs/functions.php: 499 EnsureEnabled)(/plugins/routerconfigs/classes/PHPConnection.php: 134 DoCommand)(/plugins/routerconfigs/classes/PHPConnection.php: 229 GetResponse)(/plugins/routerconfigs/classes/PHPConnection.php: 284 CactiErrorHandler)(/lib/functions.php: 4590 cacti_debug_backtrace)
2019/02/19 15:47:18 - ERROR PHP NOTICE in Plugin 'routerconfigs': Undefined index: AnyKey in file: /usr/share/cacti/plugins/routerconfigs/classes/PHPConnection.php on line: 284
2019/02/19 15:47:18 - CMDPHP PHP ERROR NOTICE Backtrace: (/plugins/routerconfigs/router-download.php: 133 plugin_routerconfigs_download)(/plugins/routerconfigs/functions.php: 159 plugin_routerconfigs_download_config)(/plugins/routerconfigs/functions.php: 499 EnsureEnabled)(/plugins/routerconfigs/classes/PHPConnection.php: 134 DoCommand)(/plugins/routerconfigs/classes/PHPConnection.php: 229 GetResponse)(/plugins/routerconfigs/classes/PHPConnection.php: 284 CactiErrorHandler)(/lib/functions.php: 4590 cacti_debug_backtrace)
2019/02/19 15:47:18 - ERROR PHP NOTICE in Plugin 'routerconfigs': Undefined index: AnyKey in file: /usr/share/cacti/plugins/routerconfigs/classes/PHPConnection.php on line: 284
2019/02/19 15:47:18 - CMDPHP PHP ERROR NOTICE Backtrace: (/plugins/routerconfigs/router-download.php: 133 plugin_routerconfigs_download)(/plugins/routerconfigs/functions.php: 159 plugin_routerconfigs_download_config)(/plugins/routerconfigs/functions.php: 499 EnsureEnabled)(/plugins/routerconfigs/classes/PHPConnection.php: 134 DoCommand)(/plugins/routerconfigs/classes/PHPConnection.php: 229 GetResponse)(/plugins/routerconfigs/classes/PHPConnection.php: 284 CactiErrorHandler)(/lib/functions.php: 4590 cacti_debug_backtrace)
2019/02/19 15:47:18 - ERROR PHP NOTICE in Plugin 'routerconfigs': Undefined index: AnyKey in file: /usr/share/cacti/plugins/routerconfigs/classes/PHPConnection.php on line: 284
2019/02/19 15:47:18 - RCONFIG 10.234.254.14 (SSH) -> DEBUG: -->
2019/02/19 15:47:18 - RCONFIG 10.234.254.14 (SSH) -> NOTICE: Ensuring process is enabled
2019/02/19 15:47:18 - RCONFIG 10.234.254.14 (SSH) -> DEBUG: Connected via ssh
```

_Originally posted by @wazwz in https://github.com/Cacti/plugin_routerconfigs/issues/87#issuecomment-465159629_

**Comment by netniV** [comment]
@wazwz This is now in the main development code, please check and close if issue is resolved.

**Comment by wazwz** [comment]
Ok, The field anykey is still missing in the table (plugin_routerconfigs_devicetypes) even, when I uninstall and Install the plugin. Do you have any recommandations / best practices to upgrade the plugin.

Thanks.

**Comment by netniV** [comment]
Does your version correctly show 1.4.2?

**Comment by wazwz** [comment]
No, It shows 1.4.1

**Comment by netniV** [comment]
Download the latest develop version and update.

**Comment by netniV** [comment]
@wazwz 	did that resolve things for you?

**Comment by netniV** [comment]
This should now be resolved as of #95
